### DISCUSSION TITLE
Use of useQuery causes 4 renders

### DISCUSSION THREAD

**Discussion by benbankes** [TanStack/query#685] [General] [answered]
Simply using `useQuery` in a component causes the component to be rendered four times.  Why?

I would expect it to only render 3 times:
1. Initial render
1. After the request
1. After the response

Why four?  Is that intended, or is it a bug?

I have created a codesandbox to demonstrate.
https://codesandbox.io/s/react-query-4x-render-hsdwb

**Comment by tannerlinsley** [comment, marked as answer]
The fourth render is the query being marked as `stale`.

**Reply by benbankes** [reply]
Ah.
